package/Shaders/Utility.hlsl

```diff
@@ -363,1037 +364,52 @@ cbuffer AlphaTestRefCB : register(b11)
-		{ 0.236732, 0.134098 },
-		{ 0.987640, 0.088839 },
-		{ 0.818842, 0.427076 },
-		{ 0.110019, 0.473128 },
-		{ 0.725730, 0.838618 },
-		{ 0.752953, 0.201086 },
-		{ 0.549883, 0.498978 },
-		{ 0.399152, 0.783258 },
-		{ 0.449843, 0.307382 },
-		{ 0.303385, 0.014557 },
-		{ 0.442976, 0.371654 },
-		{ 0.449446, 0.846675 }
+		float2(0.105594f, 0.25663f) * 2.0 - 1.0,
+		float2(0.874569f, 0.149052f) * 2.0 - 1.0,
+		float2(0.467879f, 0.644795f) * 2.0 - 1.0,
+		float2(0.00152593f, 0.856075f) * 2.0 - 1.0,
+		float2(0.85107f, 0.984954f) * 2.0 - 1.0,
+		float2(0.967528f, 0.541826f) * 2.0 - 1.0,
+		float2(0.494308f, 0.155065f) * 2.0 - 1.0,
+		float2(0.35902f, 0.954466f) * 2.0 - 1.0
 	};
 
 	float visibility = 0;
 	for (int sampleIndex = 0; sampleIndex < SampleCount; ++sampleIndex) {
-		float sampleOffsetScaleMultiplier = !asymmetric;
-		if (asymmetric) {
-			float maxSampleOffsetY = max(abs(2 * PoissonDiskSampleOffsets[2 * sampleIndex].y + sampleOffsetShift), abs(2 * PoissonDiskSampleOffsets[2 * sampleIndex + 1].y + sampleOffsetShift));
-			float uvScale = -1;
-			float uvThreshold = -0.5;
-			if (baseUv.y >= 0.5) {
-				uvScale = 1;
-				uvThreshold = 0.5;
-			}
-			if (uvScale * (baseUv.y + maxSampleOffsetY) >= uvThreshold) {
-				sampleOffsetScaleMultiplier = 1;
-			}
-		}
-		float2 sampleOffset1 = 2 * PoissonDiskSampleOffsets[2 * sampleIndex] + sampleOffsetShift;
-		float2 sampleOffset2 = 2 * PoissonDiskSampleOffsets[2 * sampleIndex + 1] + sampleOffsetShift;
-		float2 sampleOffsetScale = float2(ShadowSampleParam.z, ShadowSampleParam.z * sampleOffsetScaleMultiplier);
-		float2 sampleUv1 = sampleOffset1 * sampleOffsetScale + baseUv;
-		float2 sampleUv2 = sampleOffset2 * sampleOffsetScale + baseUv;
-		visibility += tex.SampleCmpLevelZero(samp, float3(sampleUv1, layerIndex), compareValue).x;
-		visibility += tex.SampleCmpLevelZero(samp, float3(sampleUv2, layerIndex), compareValue).x;
+		float2 sampleOffset = mul(PoissonDiskSampleOffsets[sampleIndex], rotationMatrix);
+		float2 sampleOffsetScale = ShadowSampleParam.z;
+
+#	if defined(RENDER_SHADOWMASKDPB)
+		float2 sampleUV = sampleOffset * 2 + baseUV;
+
+		baseUV.z += noise;
+
+		bool lowerHalf = baseUV.z * 0.5 + 0.5 < 0;
+		float3 normalizedPositionLS = normalize(float3(sampleUV.xy, baseUV.z));
+
+		float3 positionOffset = lowerHalf ? float3(0, 0, -1) : float3(0, 0, 1);
+		float3 lightDirection = normalize(normalizedPositionLS + positionOffset);
+		float2 shadowMapUV = lightDirection.xy / lightDirection.z * 0.5 + 0.5;
+		shadowMapUV.y = lowerHalf ? 1 - 0.5 * shadowMapUV.y : 0.5 * shadowMapUV.y;
+
+		visibility += tex.SampleCmpLevelZero(samp, float3(shadowMapUV, layerIndex), compareValue + noise * 0.001).x;
+
+#	else
+		float2 sampleUV = sampleOffset * sampleOffsetScale + baseUV;
+		visibility += tex.SampleCmpLevelZero(samp, float3(sampleUV, layerIndex), compareValue + noise * 0.001).x;
+#	endif
 	}
-	return visibility / (2 * SampleCount);
+	return visibility / float(SampleCount);
```

Comment:
Since it's const u should have a const for rcp too

Reply:
the compiler can handle this